### PR TITLE
test(extension): refactor AddressInput POM class

### PR DESCRIPTION
**Pull request by wklos-iohk**
# Checklist

- [X] JIRA - LW-9754
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

### CODE REVIEW

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts [hunk 1]
```diff
@@ -91,16 +91,16 @@ When(
         addr = address;
         break;
     }
-    await transactionExtendedPageObject.fillAddress(addr, inputIndex);
+    await new AddressInput(String(inputIndex).length === 0 ? 1 : inputIndex).fillAddress(addr);
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts:94]: This logic can be moved inside Address Input definition and duplications can be removed
`String(inputIndex).length === 0 ? 1 : inputIndex`

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts [hunk 2]
```diff
@@ -91,16 +91,16 @@ When(
...
   }
 );
 
 When(/I enter "([^"]*)" in the bundle (\d) recipient's address/, async (value: string, inputIndex: number) => {
-  await transactionExtendedPageObject.fillAddress(value, inputIndex);
+  await new AddressInput(String(inputIndex).length === 0 ? 1 : inputIndex).fillAddress(value);
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts:99]: as above

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts [hunk 3]
```diff
@@ -157,13 +157,14 @@ Then(
 Then(
   /I enter an address (\d*) that matches the amount of characters but does not match with the checksum/,
   async (inputIndex?: number) => {
-    await transactionExtendedPageObject.fillAddress(shelleyInvalid.getAddress(), inputIndex);
+    await new AddressInput(String(inputIndex).length === 0 ? 1 : inputIndex).fillAddress(shelleyInvalid.getAddress());
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts:160]: as above

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts [hunk 4]
```diff
@@ -202,19 +203,19 @@ Then(/^the balance of token is displayed in coin selector$/, async () => {
 });
 
 Then(/^click "(Add|Remove) address" button (\d*) in address bar$/, async (_ignored: string, inputIndex: number) => {
-  await new AddressInput(inputIndex).searchLoader.waitForDisplayed({ reverse: true });
-  await transactionExtendedPageObject.clickAddAddressButton(inputIndex);
+  const addressInput = new AddressInput(String(inputIndex).length === 0 ? 1 : inputIndex);
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts:206]: as above

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts [hunk 5]
```diff
@@ -202,19 +203,19 @@ Then(/^the balance of token is displayed in coin selector$/, async () => {
...
+  await addressInput.searchLoader.waitForDisplayed({ reverse: true });
+  await addressInput.clickAddAddressButton();
 });
 
 When(
   /^I fill bundle (\d+) with "([^"]*)" address with following assets:$/,
   async (bundleIndex, receivingAddress, options) => {
     await browser.pause(500);
-    await transactionExtendedPageObject.fillAddress(
+    await new AddressInput(String(bundleIndex).length === 0 ? 1 : bundleIndex).fillAddress(
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts:215]: as above

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts [hunk 6]
```diff
@@ -627,7 +627,7 @@ Then(/^recipients address input contains address entry with name "([^"]*)"$/, as
 });
 
 Then(/^recipients address input (\d*) is empty$/, async (inputIndex: number) => {
-  await drawerSendExtendedAssert.assertSeeEmptyRecipientsAddressInput(inputIndex);
+  await drawerSendExtendedAssert.assertSeeEmptyRecipientsAddressInput(String(inputIndex).length === 0 ? 1 : inputIndex);
```
Comment (bslabiak) [inline packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts:630]: as above

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts (hunk 5 above)
Comment (bslabiak) [reply packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts:218]: One left over

**Review by bslabiak — APPROVED** [review]

File: packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts (hunk 5 above)
Comment (wklos-iohk) [reply packages/e2e-tests/src/steps/sendTransactionSimpleSteps.ts]: updated

**Pull request merged**